tests/ui/suggestions/remove-brackets-range-issue-147944.rs

```diff
@@ -0,0 +1,27 @@
+fn main() {
+    let mut vec = vec![1, 2, 3, 4, 5];
+    vec.drain([..3]);
+    //~^ ERROR: the trait bound `[RangeTo<{integer}>; 1]: RangeBounds<usize>` is not satisfied [E0277]
+    //~| SUGGESTION: ..3
+
+    // Should not produce this suggestion:
+
+    let range = [1..5];
+    let mut vec = vec![1, 2, 3, 4, 5];
+    vec.drain(range);
+    //~^ ERROR: the trait bound `[std::ops::Range<{integer}>; 1]: RangeBounds<usize>` is not satisfied [E0277]
```

Comment:
> I edited the tests to include this example:
> 
> ```rust
> let range = [1..5];
> let mut vec = vec![1, 2, 3, 4, 5];
> vec.drain(range);
> ```
>
> And I have a question about this. Should compiler give an error in this case or is it outside of a scope for such suggestion?

I think it's fine to leave this as future work, maybe with the note like this instead:

```suggestion
    // Does not produce this suggestion (but could in the future):

    let range = [1..5];
    let mut vec = vec![1, 2, 3, 4, 5];
    vec.drain(range);
    //~^ ERROR: the trait bound `[std::ops::Range<{integer}>; 1]: RangeBounds<usize>` is not satisfied [E0277]
    
    // Should not produce this suggestion:
```